### PR TITLE
Add org.eu.encom.matrique

### PR DESCRIPTION
**Pull request by encombhat**
Add Flatpak for Matrique.

### CODE REVIEW

File: matrique.appdata.xml [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+  <launchable type="desktop-id">org.eu.encom.matrique</launchable>
+  <url type="homepage">https://gitlab.com/b0/matrique</url>
+  <content_rating type="oars-1.0">
+    <content_attribute id="social-chat">intense</content_attribute>
+    <content_attribute id="social-audio">intense</content_attribute>
+  </content_rating>
+  <releases>
+    <release version="167" date="2018-08-14" type="stable"></release>
+  </releases>
+</component>
```
Comment (eszlari) [inline matrique.appdata.xml]: You have to add at least one screenshot.

File: org.eu.encom.matrique.json [hunk 2]
```diff
@@ -0,0 +1,53 @@
+
+{
+    "id": "org.eu.encom.matrique",
+    "rename-icon": "matrique",
+    "rename-desktop-file": "matrique.desktop",
+    "rename-appdata-file": "matrique.appdata.xml",
+    "runtime": "org.kde.Platform",
+    "runtime-version": "5.10",
```
Comment (eszlari) [inline org.eu.encom.matrique.json]: 5.10 is not maintained anymore, use 5.9 or 5.11.

File: org.eu.encom.matrique.json [hunk 3]
```diff
@@ -0,0 +1,53 @@
...
+        "--device=dri",
+	"--filesystem=home:rw",
+	"--filesystem=/tmp:rw",
+	"--talk-name=org.freedesktop.Notifications",
+	"--talk-name=org.kde.StatusNotifierWatcher"
+    ],
+    "modules": [
+        {
+            "name": "matrique",
+            "buildsystem": "simple",
```
Comment (eszlari) [inline org.eu.encom.matrique.json]: Use` buildsystem: qmake` and `post-install`.

**Comment by encombhat** [comment]
@eszlari Should be fixed by now.

**Comment by eszlari** [comment]
Why don't you use the icons and .desktop file from the source tree?

And since you seem to be upstream, please add the appdata.xml file to the source tree. And install all files through qmake / make install, see e.g.:
https://github.com/mltframework/shotcut/blob/master/src/src.pro#L355

File: matrique.appdata.xml [hunk 4]
```diff
@@ -0,0 +1,43 @@
...
+​      <image>https://raw.githubusercontent.com/encombhat/flathub/org.eu.encom.matrique/screenshots/4.png</image>
+​    </screenshot>
+​  </screenshots>
+  <url type="homepage">https://gitlab.com/b0/matrique</url>
+  <content_rating type="oars-1.0">
+    <content_attribute id="social-chat">intense</content_attribute>
+    <content_attribute id="social-audio">intense</content_attribute>
+  </content_rating>
+  <releases>
+    <release version="167" date="2018-08-14" type="stable"></release>
```
Comment (eszlari) [inline matrique.appdata.xml]: git tag?

File: org.eu.encom.matrique.json [hunk 5]
```diff
@@ -0,0 +1,37 @@
+
+{
+    "id": "org.eu.encom.matrique",
+    "rename-icon": "matrique",
+    "rename-desktop-file": "matrique.desktop",
+    "rename-appdata-file": "matrique.appdata.xml",
```
Comment (eszlari) [inline org.eu.encom.matrique.json]: You can rename these files in the upstream source tree.

File: org.eu.encom.matrique.json [hunk 6]
```diff
@@ -0,0 +1,37 @@
...
+    "sdk": "org.kde.Sdk",
+    "command": "matrique",
+    "finish-args": [
+        "--share=ipc",
+        "--share=network",
+        "--socket=x11",
+        "--socket=wayland",
+        "--device=dri",
+	"--filesystem=home:rw",
+	"--filesystem=/tmp:rw",
```
Comment (eszlari) [inline org.eu.encom.matrique.json]: "rw" can be dropped

**Comment by eszlari** [comment]
This is not necessary, but have you considered using YAML instead of JSON?

**Comment by encombhat** [comment]
I never know manifest can be YAML. Any document of this?

**Comment by encombhat** [comment]
@eszlari Anyway I fixed these issues and changed the manifest to YAML.

**Comment by eszlari** [comment]
Please fix the indentation: https://github.com/flathub/flathub/wiki/YAML-Style-Guide

**Comment by encombhat** [comment]
@eszlari I fixed the indentation.

File: org.eu.encom.matrique.yaml [hunk 7]
```diff
@@ -0,0 +1,24 @@
...
+sdk: org.kde.Sdk
+command: matrique
+finish-args:
+  - '--share=ipc'
+  - '--share=network'
+  - '--socket=x11'
+  - '--socket=wayland'
+  - '--device=dri'
+  - '--filesystem=home'
+  - '--filesystem=/tmp'
```
[the remark this answers is not in the document]
Comment (encombhat) [reply org.eu.encom.matrique.yaml]: Qt's openUrlExternally() does not work if it uses tmp directory in the sandbox, and I didn't find a solution to this issue other than this.

[the remark this answers is not in the document]
Comment (nedrichards) [reply org.eu.encom.matrique.yaml]: Oh interesting, is that known @grulja ?

File: org.eu.encom.matrique.yaml [hunk 8]
```diff
@@ -0,0 +1,24 @@
...
+  - '--talk-name=org.freedesktop.Notifications'
+  - '--talk-name=org.kde.StatusNotifierWatcher'
+modules:
+  - name: matrique
+    buildsystem: qmake
+    sources:
+      - type: git
+        url: 'https://gitlab.com/b0/matrique'
+        tag: 167
```
[the remark this answers is not in the document]
Comment (encombhat) [reply org.eu.encom.matrique.yaml]: Well, the tag is literally "167".

**Comment by encombhat** [comment]
I changed the quoting.

File: org.eu.encom.matrique.yaml [hunk 9]
```diff
@@ -0,0 +1,24 @@
...
+sdk: org.kde.Sdk
+command: matrique
+finish-args:
+  - --share=ipc
+  - --share=network
+  - --socket=x11
+  - --socket=wayland
+  - --device=dri
+  - --filesystem=home
+  - --filesystem=/tmp
```
Comment (eszlari) [inline org.eu.encom.matrique.yaml]: Does it work, if you replace this with `--env=TMPDIR=$XDG_CACHE_HOME`?

File: org.eu.encom.matrique.yaml [hunk 10]
```diff
@@ -0,0 +1,24 @@
...
+  - --talk-name=org.freedesktop.Notifications
+  - --talk-name=org.kde.StatusNotifierWatcher
+modules:
+  - name: matrique
+    buildsystem: qmake
+    sources:
+      - type: git
+        url: 'https://gitlab.com/b0/matrique'
```
Comment (eszlari) [inline org.eu.encom.matrique.yaml]: Quotation can be removed.

File: org.eu.encom.matrique.yaml (hunk 9 above)
Comment (encombhat) [reply org.eu.encom.matrique.yaml]: Sorry but that doesn't work. The file can't even write to $XDG_CACHE_HOME dir.

Comment (encombhat) [reply org.eu.encom.matrique.yaml]: Will it work if I change upstream code and let file download in "~/.cache/<APPNAME>" instead?

Comment (encombhat) [reply org.eu.encom.matrique.yaml]: It worked. I don't believe it.

**Comment by barthalion** [comment]
bot, build org.eu.encom.matrique

**Comment by encombhat** [comment]
Looks like the build succeeded?

**Comment by barthalion** [comment]
Repository has been created: https://github.com/flathub/org.eu.encom.matrique

Thanks!

**Comment by encombhat** [comment]
@barthalion The build failed at cloning git repo and it seems like a network issue.
Can you trigger a rebuild? Thanks.

**Comment by encombhat** [comment]
@barthalion It seems that I don't have write access to https://github.com/flathub/org.eu.encom.matrique .
What happened?
image

**Comment by barthalion** [comment]
You should receive an invitation to collaboration on the repository. It has be accepted before write access is granted.

**Comment by encombhat** [comment]
Oh, I thought the invitation is sent to Github notification and didn't check the email. 

Gitlab.com works for me, and the build from YAML succeeded on my computer.
I guess something is wrong with the builder?

**Comment by encombhat** [comment]
Looks like the builder fails if the Git repo does not end with ".git"?
I need to change all my submodules then.

**Comment by barthalion** [comment]
Yes, some builders are running ancient git so it might be making a difference.

**Pull request closed without merging**